sdcm/cluster_aws.py

```diff
@@ -683,16 +683,22 @@ def traffic_control(self, tcconfig_params=None):
         :param tcconfig_params: commandline arguments for tcset, if None will call tcdel
         :return: None
         """
-
-        self.remoter.run("sudo modprobe sch_netem")
-
         if tcconfig_params is None:
             tc_command = LOCAL_CMD_RUNNER.run("tcdel eth1 --tc-command", ignore_status=True).stdout
             self.remoter.run('sudo bash -cxe "%s"' % tc_command, ignore_status=True)
         else:
             tc_command = LOCAL_CMD_RUNNER.run("tcset eth1 {} --tc-command".format(tcconfig_params)).stdout
             self.remoter.run('sudo bash -cxe "%s"' % tc_command)
 
+    def install_traffic_control(self):
+        self.remoter.run("sudo modprobe sch_netem")
```

Comment:
why did you move the modprobe here? the node can be restarted in between network nemesis and the module won't be loaded.

Reply:
it runs 3 times for 1 nemesis each time when the traffic_control is called.
I call install_traffic_control each time in nemesis at the beginning

Reply:
It sounds safe to load the module by modprobe everytime.

But we only need to prepare once for the whole job with following steps:

- load the module, make it work before restart
    - sudo modprobe sch_netem
- install tc
- set to load module after restart
    - echo sch_netem |sudo tee /etc/modules-load.d/tc.conf

Reply:
@amoskong why it could be safer? the nemesises which use it are disruptive, so it will not be running in parallel with any reboot/hard reboot nemesis. So i think it will be enough to load it on nemesis start.

Reply:
> @amoskong why it could be safer? the nemesises which use it are disruptive, so it will not be running in parallel with any reboot/hard reboot nemesis. So i think it will be enough to load it on nemesis start.

Your current solution is safe.

Reply:
the simplest solution is to keep the module loading in the place where it was before traffic_control(). Why don't you want to keep it there?

Reply:
The idea was to have all commands related to installation in one function.
Traffic_control is called 3 times during one nemesis, So i didn't see requirement to execute command to load module each time.

Return as you recommended